### ISSUE TITLE
Not possible to switch to creating "Text" stories using VoiceOver

### ISSUE THREAD

**Issue by awaitlink** [signalapp/Signal-iOS#5744]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

---

### Bug description

It's not possible to switch to creating "Text" stories using VoiceOver.

Original report on the forum: https://community.signalusers.org/t/accessibility-of-stories-creation-with-voiceover/58617.

### Steps to reproduce
1. Have VoiceOver enabled.
2. Go to Stories tab > Camera icon > select the "Camera / Text" control and attempt to change it to "Text".

**Actual result:** The "Text" text is selected visually, however the composer doesn't switch to text mode.

**Expected result:** It should be possible to switch composer modes using VoiceOver.

### Device info
**Device**: iPhone 12 mini
**iOS version**: iOS 17.3 (21D50)
**Signal version:** 6.59.0.12 (latest beta)

### Link to debug log

https://debuglogs.org/ios/6.59.0/2498abe062a04008288c1e2b00deb7cc2ca2f71a586ca2e699e396910995b2d9.zip
